### PR TITLE
feat(datetimerangepicker): presented as form control

### PR DESCRIPTION
**Pull request by alebinson**
Related Issues:
https://github.com/MapColonies/shared-components/issues/63
https://github.com/MapColonies/discrete-layer-client/issues/3
Closes issues: 
https://github.com/MapColonies/shared-components/issues/63

## Checklist
- [x] Tests
- [x] Stories
- [x] Lint
- [x] Prettier

## What I did
Added a new control which gives an ability to present a datetimepicker as form control or button
- [ ] BREAKING CHANGE
- [x] feature request
- [ ] bug fix
- [x] documentation

More information:

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -81,3 +82,15 @@ or
 ```sh
   yarn run storybook-composition
 ``` 
+
+In order to run storybook for specific package - run it from appropriate package location
+1. react-components
+```
```
Comment (syncush) [inline README.md]: you can use ```sh, for syntax highlight

File: README.md [hunk 2]
```diff
@@ -81,3 +82,15 @@ or
...
+  cd packages/react-components
+  yarn run storybook
+```
+2. react-ui-components
+```
```
Comment (syncush) [inline README.md]: same

**Review by syncush — CHANGES_REQUESTED** [review]
1. From a quick look from the storybook, I can not set a future date in `to` date picker.

2. `Date time range looks like input` story is glitched when setting `from` and `to` dates, the text of the dates overlap with the `date_range` text.

3. I think we should add some features and introduce new properties for existing date picker component:
- Explicit `min` and `max` dates.
- Date with time or without time

File: README.md (hunk 1 above)
Comment (alebinson) [reply README.md]: Done

File: README.md (hunk 2 above)
Comment (alebinson) [reply README.md]: Done

**Comment by alebinson** [comment]
1. From a quick look from the storybook, I can not set a future date in `to` date picker.
ALEX --> Property **disableFuture** added to components API in order to able future time selection control

2. `Date time range looks like input` story is glitched when setting `from` and `to` dates, the text of the dates overlap with the `date_range` text.
ALEX --> It's an icon usage when no Icon find icon name is shown

3. I think we should add some features and introduce new properties for existing date picker component:
- Explicit `min` and `max` dates.
ALEX --> **minDate** and **maxDate** added as API props
- Date with time or without time
ALEX --> lets do it in other issue, a bit expansive

**Comment by alebinson** [comment]
Compnent usage should be like this 
MicrosoftTeams-image

MicrosoftTeams-image (1)

**Comment by syncush** [comment]
@alebinson 
please open an issue for disabling \ enabling time and mention this PR.

File: packages/react-components/src/lib/date-range-picker/stories/DateRangePicker.stories.tsx [hunk 3]
```diff
@@ -15,3 +18,182 @@ export const DateTime: CSFStory<JSX.Element> = () => (
...
+
+DateNoFutureLimitTime.story = {
+  name: 'Date time range no future limit',
+};
+
+export const DateMinMaxLimitTime: CSFStory<JSX.Element> = () => (
+  <DateTimeRangePicker
+    onChange={action('date changed')}
+    minDate={new Date('09/01/2020')}
+    maxDate={new Date('11/01/2020')}
```
Comment (syncush) [inline packages/react-components/src/lib/date-range-picker/stories/DateRangePicker.stories.tsx]: use `current date` for minDate and `new date + 5 days` so it is relevant for every date in the future.

**Review by syncush — CHANGES_REQUESTED** [review]

File: packages/react-components/src/lib/date-range-picker/stories/DateRangePicker.stories.tsx (hunk 3 above)
Comment (alebinson) [reply packages/react-components/src/lib/date-range-picker/stories/DateRangePicker.stories.tsx]: Done, added +/- delta for current date for min and max dates

**Comment by alebinson** [comment]
Opened https://github.com/MapColonies/shared-components/issues/67 in order to cover timeless mode

**Review by syncush — APPROVED** [review]

**Pull request merged**